From 3f7dd82b90b29bb1ad701d82377a462d6fd80d16 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 13 May 2021 21:53:45 +0200
Subject: [PATCH] Remove the usage of deprecated
 'xml.etree.ElementTree.Element.getchildren'

This fix is needed for Python 3.9.
---
 src/presets.py | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/presets.py b/src/presets.py
index 2cc8cf5..40a6f48 100644
--- a/src/presets.py
+++ b/src/presets.py
@@ -271,7 +271,7 @@ def _load_author(root):
     """
     author = Author()
     
-    for child in root.getchildren():
+    for child in list(root):
         if child.tag == "name":
             author.name = child.text.strip()
         elif child.tag == "email":
@@ -290,7 +290,7 @@ def _load_audio_codec(root):
     """
     codec = AudioCodec()
     
-    for child in root.getchildren():
+    for child in list(root):
         if child.tag == "name":
             codec.name = child.text.strip()
         elif child.tag == "container":
@@ -304,7 +304,7 @@ def _load_audio_codec(root):
         elif child.tag == "samplerate":
             codec.samplerate = child.text.strip()
         elif child.tag == "presets":
-            for command in child.getchildren():
+            for command in list(child):
                 codec.presets.append(command.text.strip())
     
     return codec
@@ -320,7 +320,7 @@ def _load_video_codec(root):
     """
     codec = VideoCodec()
     
-    for child in root.getchildren():
+    for child in list(root):
         if child.tag == "name":
             codec.name = child.text.strip()
         elif child.tag == "container":
@@ -340,7 +340,7 @@ def _load_video_codec(root):
         elif child.tag == "passes":
             codec.passes = child.text.strip()
         elif child.tag == "presets":
-            for command in child.getchildren():
+            for command in list(child):
                 codec.presets.append(command.text.strip())
     
     return codec
@@ -356,7 +356,7 @@ def _load_preset(root):
     """
     preset = Preset()
     
-    for child in root.getchildren():
+    for child in list(root):
         if child.tag == "name":
             preset.name = child.text.strip()
         elif child.tag == "container":
@@ -391,7 +391,7 @@ def load(filename):
     
     device.filename = filename
     
-    for child in tree.getroot().getchildren():
+    for child in list(tree.getroot()):
         if child.tag == "make":
             device.make = child.text.strip()
         elif child.tag == "model":
-- 
2.31.1

